### ISSUE TITLE
[4.1] [PHP 8.1] Debug plugin: 4 type issues (in vendor/maximebf/DebugBar): 

### ISSUE THREAD

**Issue by beat** [joomla/joomla-cms#36772]
### Steps to reproduce the issue

Go PHP 8.1, Enable debug, and see in backend the warnings

### Expected result

No warnings

### Actual result

```
Deprecated: substr(): Passing null to parameter #1 ($string) of type string is deprecated in libraries/vendor/maximebf/debugbar/src/DebugBar/JavascriptRenderer.php on line 792

Deprecated: preg_match(): Passing null to parameter #2 ($subject) of type string is deprecated in libraries/vendor/maximebf/debugbar/src/DebugBar/JavascriptRenderer.php on line 792

Deprecated: strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in libraries/vendor/maximebf/debugbar/src/DebugBar/JavascriptRenderer.php on line 808

Deprecated: Implicit conversion from float -INF to int loses precision in libraries/vendor/maximebf/debugbar/src/DebugBar/DataFormatter/DataFormatter.php on line 81
```

### System information (as much as possible)

4.0.6, PHP 8.1.1

### Additional comments

This file is not in the joomla/joomla-cms repository, where is it so I can propose a MR ?

The first 2 warnings are already fixed upstream here:
https://github.com/maximebf/php-debugbar/blob/master/src/DebugBar/JavascriptRenderer.php#L792-L794
and here:
https://github.com/maximebf/php-debugbar/blob/master/src/DebugBar/JavascriptRenderer.php#L810

For the last one, an (int) needs to be added like this, as arrays can have only int and string indices and floor() returns an int encoded as a float so the indice is an int, which corresponds to the natural array above:

```php
return $sign . round(pow(1024, $base - floor($base)), $precision) . $suffixes[(int) floor($base)];
```

I have submitted a corresponding PR upstream here:
https://github.com/maximebf/php-debugbar/pull/498

**Comment by brianteeman** [comment]
> This file is not in the joomla/joomla-cms repository, where is it so I can propose a MR ?

That is correct. It is pulled in via composer. The current version as defined in the composer.lock is the latest release 

The issues already fixed there (probably [#494](https://github.com/maximebf/php-debugbar/pull/494) by @alikon just need to be "released"

**Comment by brianteeman** [comment]
In summary - you've done all you can and there is no need for an open issue here and this can be closed.

**Comment by beat** [comment]
ok, closing then.

**Comment by beat** [comment]
There are other issues that are not committed, and Joomla 4.1 will be issuing those errors if they or we don't fix them. E.g. See my PR here:
https://github.com/joomla/joomla-cms/issues/36772
What should we do ? just ship with the error, or manually fix the remaining PHP 8.1 incompatibilities during the build process ?

**Comment by laoneo** [comment]
@beat, can this one here be closed as for me it looks like a duplicate of #37274?

**Comment by Quy** [comment]
Closing per PR #37488.
